app.rb

```diff
@@ -0,0 +1,170 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  attr_reader :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'List of Books:'
+    books.each do |book|
+      puts "#{book.title} by #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'List of People:'
+    people.each do |person|
+      puts "#{person.name} (#{person.class})"
+    end
+  end
+
+  def create_person
+    puts 'Create a Person'
+    puts 'Enter name:'
+    name = gets.chomp
+
+    puts 'Enter age:'
+    age = gets.chomp.to_i
+
+    puts 'Is the person a student or a teacher? (s/t)'
+    person_type = gets.chomp.downcase
+
+    case person_type
+    when 's'
+      puts "Enter the person's classroom:"
+      classroom = gets.chomp
+
+      person = Student.new(name, age, classroom)
+      people << person
+      puts "Student '#{person.name}' created!"
+    when 't'
+      puts "Enter the person's specialization:"
+      specialization = gets.chomp
+
+      person = Teacher.new(name, age, specialization)
+      people << person
+      puts "Teacher '#{person.name}' created!"
+    else
+      puts 'Invalid person type.'
+    end
+  end
```

Comment:
Good job so far. :+1: However, the issue with creating a person still persist when trying to create a person (student or teacher) as shown below. Kindly check the number of arguments the student and teacher classes are expected to take (as well as the classroom class) to fix the errors.

![Screenshot from 2023-06-30 17-35-37](https://github.com/Bellagirl-maker/oop-school-library/assets/65233860/b8b80130-9161-404f-bdb3-ac8019f8058f)

![Screenshot from 2023-06-30 17-35-23](https://github.com/Bellagirl-maker/oop-school-library/assets/65233860/8343d5f6-ca8a-4806-98e3-0e4cb9f5041a)

Reply:
@jerryowusu done with this change